CRM/Member/BAO/Membership.php

```diff
@@ -2508,4 +2514,93 @@ public static function getContactsCancelledMembership($contactID, $isTest = FALS
+   * priceset, unlike online renewals which only renew for a specific price set
+   *
+   * @param type $contactId
+   * @return type
+   *
+   */
+  public static function getContactMemberhipsByMembeshipOrg($contactId) {
+    $query = "
+      select distinct mem.membership_type_id, mem.id as membership_id, t.member_of_contact_id, co.receive_date, s.is_current_member, li.contribution_id
+ from civicrm_membership mem
+ 		inner join civicrm_membership_type t on t.id = mem.membership_type_id
+ 		inner join civicrm_membership_status s on s.id = mem.status_id
+ 		left join civicrm_line_item li on li.entity_table = 'civicrm_membership' and li.entity_id = mem.id
+ 		left join civicrm_contribution co on co.id = li.contribution_id
+  where mem.contact_id = $contactId
+  order by t.member_of_contact_id,
+           case when s.is_current_member then 0 else 1 end,
+  	      co.receive_date desc
+    ";
+
+    $toReturn = array();
+    $dao = CRM_CORE_DAO::executeQuery($query);
+    while ($dao->fetch()) {
+      if (key_exists($dao->member_of_contact_id, $toReturn)) {
+        continue;
+      }
+      $toReturn[$dao->member_of_contact_id] = array(
+        'membership_type_id' => $dao->membership_type_id,
+        'receive_date' =>$dao->receive_date,
+        'is_current_member' =>$dao->is_current_member,
+        'membership_id' =>$dao->membership_id
+      );
+    }
+    return $toReturn;
+
+  }
+
+  /**
+   *
+   * Given specified membership types, returns the membership ids that are applicable
+   * (if one exists) for the contact.
+   *
+   * @param type $contactId
+   *  e.g., 3.
+   * @param type $membershipTypes
+   *  e.g., 6, 8
+   *
+   * @return array (6 => 10, 8 => NULL) to indicate that
+   *   contact #3's membership type id 6 points to an org that already has a membership
+   *   for this contact (and that memebership id happens to be 10.  In the case of
+   *   membership type 8, the contact does not have a membership for this org (hence NULL)
+   *
+   */
+  public static function getMembershipIdsForMembershipTypeOrgs($contactId, $membershipTypes) {
+    $query = "
+          select id, member_of_contact_id, (
+                select id
+                  from civicrm_membership mem
+                 where contact_id = $contactId
+                   and membership_type_id in (select id from civicrm_membership_type mt2 where mt2.member_of_contact_id = mt1.member_of_contact_id)
+                 order by
+                    case when mem.status_id in (select id from civicrm_membership_status where is_current_member)
+                         then 1
+                         else 0
+                    end,
+                    end_date DESC
+                limit 1) existing_membership_id
+            from civicrm_membership_type mt1
+           where id in (" . implode(", ", array_keys($membershipTypes) ) . ");
+    ";
+    $dao = CRM_Core_DAO::executeQuery($query);
```

Comment:
Same would apply here. but this time you probably want string for membershipTypes